File: EHealthMonitoringSystemBackend/EHealthMonitoringSystemBackend.Data/Models/PatientProfile.cs
using System.ComponentModel.DataAnnotations;

namespace EHealthMonitoringSystemBackend.Core.Models;

public class PatientProfile
{
    [Key]
    public int Id { get; set; }

    [Required]
    public string? FirstName { get; set; }

    [Required]
    public string? LastName { get; set; }

    [Required]
    public string? PhoneNumber { get; set; }

    [Required]
    public string? Cnp { get; set; }
}

File: e_health_monitoring_system_frontend/lib/models/patient_profile.dart
import 'dart:convert';

class PatientProfile {
  final String firstName;
  final String lastName;
  final String phoneNumber;
  final String cnp;

  PatientProfile({
    required this.lastName,
    required this.phoneNumber,
    required this.cnp,
    required this.firstName,
  });

  String toJson() {
    return jsonEncode({
      "firstName": firstName,
      "lastName": lastName,
      "phoneNumber": phoneNumber,
      "cnp": cnp,
    });
  }

  factory PatientProfile.fromJson(String json) {
    var map = jsonDecode(json);
    return PatientProfile(
      firstName: map['firstName'],
      lastName: map['lastName'],
      cnp: map['cnp'],
      phoneNumber: map['phoneNumber'],
    );
  }
}
